### PR TITLE
5-RML-Reification-Statements

### PR DESCRIPTION
**Pull request by jhculb**
WIP: No work

### CODE REVIEW

File: src/rml-reified/bnode-1.rml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+:firstTM a rml:TriplesMap ;
+    rml:logicalSource [
+        rml:source "data.csv";
+        rml:referenceFormulation ql:CSV
+    ];
+    rml:subjectMap [
+        rr:reference "myFirstVariable" ;
+        rr:termType rr:BlankNode
+    ];
+    4
```
Comment (dachafra) [inline src/rml-reified/bnode-1.rml:17]: ups, and this 4?

File: src/rml-reified/bnode-1.rml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+@prefix ex: <http://example/> .
+@prefix : <http://example.org> .
+@base <http://example.org> .
+:firstTM a rml:TriplesMap ;
+    rml:logicalSource [
+        rml:source "data.csv";
+        rml:referenceFormulation ql:CSV
+    ];
+    rml:subjectMap [
+        rr:reference "myFirstVariable" ;
```
Comment (dachafra) [inline src/rml-reified/bnode-1.rml:14]: @andimou how do we manage this? As we shouldn't have this reference for the reification standard blank node in the data... Is there any automatic generation for BN in RML?

File: src/rml-reified/bnode-1.rml (hunk 1 above)
Comment (jhculb) [reply src/rml-reified/bnode-1.rml:17]: spolling mistake

**Comment by jhculb** [comment]
Good evening @dachafra @andimou, hope you are well. Could you review and approve the merge when you are next free please!

File: src/rml-reified/nested-1.ttl [hunk 3]
```diff
@@ -0,0 +1,91 @@
+@prefix rr: <http://www.w3.org/ns/r2rml#> .
+@prefix rml: <http://semweb.mmlab.be/ns/rml#> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix ql: <http://semweb.mmlab.be/ns/ql#> .
+@prefix ex: <http://example/> .
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#>
```
Comment (jhculb) [inline src/rml-reified/nested-1.ttl:6]: I realise these are extraneous prefixes. I assume it's fine keeping them around

Comment (dachafra) [reply src/rml-reified/nested-1.ttl:6]: this is good!

File: src/rml-reified/syntax-2.ttl [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        ]
+    ];
+
+    rr:predicateObjectMap [
+        rr:predicate rdf:object ;
+        rml:objectMap [
+            rr:template "http://example/{c3}"
+        ]
+    ];
+
```
Comment (dachafra) [inline src/rml-reified/syntax-2.ttl:46]: did you validate this RDF? I think you need a `.` at the end

Comment (dachafra) [reply src/rml-reified/syntax-2.ttl:46]: review in the rest please

File: src/rml-reified/syntax-3.ttl [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        rr:predicate rdf:type ;
+        rml:objectMap [
+            rr:reference rdf:Statement
+        ]
+    ];
+
+    rr:predicateObjectMap [
+        rr:predicate rdf:subject ;
+        rml:objectMap [
+            rml:quotedTriplesMap :elementaryTM1
```
Comment (dachafra) [inline src/rml-reified/syntax-3.ttl:105]: NO! You cannot use this in plan RML, there should be another way ;-)

File: src/rml-reified/syntax-3.ttl [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        rr:predicate rdf:predicate ;
+        rml:objectMap [
+            rr:constant ex:q
+        ]
+    ];
+
+    rr:predicateObjectMap [
+        rr:predicate rdf:object ;
+        rml:objectMap [
+            rml:quotedTriplesMap :elementaryTM2
```
Comment (dachafra) [inline src/rml-reified/syntax-3.ttl:119]: again, this is not good

File: src/rml-reified/syntax-4.ttl [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        rr:predicate rdf:predicate ;
+        rml:objectMap [
+            rr:constant ex:q
+        ]
+    ];
+
+    rr:predicateObjectMap [
+        rr:predicate rdf:object ;
+        rml:objectMap [
+            rml:quotedTriplesMap :elementaryTM2
```
Comment (dachafra) [inline src/rml-reified/syntax-4.ttl:119]: again, we cannot use this!

**Review by dachafra — CHANGES_REQUESTED** [review]
You cannot use rml:quotedTriplesMaps in plan RDF, you may need to take a look to RML spec to see how you can do it (maybe just repeating some rules...)

File: src/rml-reified/syntax-2.ttl (hunk 4 above)
Comment (jhculb) [reply src/rml-reified/syntax-2.ttl:46]: > did you validate this RDF? I think you need a `.` at the end

I did when parsing the triples as part of 6.

**Comment by jhculb** [comment]
> You cannot use rml:quotedTriplesMaps in plan RDF, you may need to take a look to RML spec to see how you can do it (maybe just repeating some rules...)

Are you referring to parentTriplesMap? This seems to be how to map two logical sources (csvs, etc.) together?

(currently reading R2RML spec to see if I've missed something there)

**Comment by dachafra** [comment]
It could be a good option, yes ;-)

**Comment by jhculb** [comment]
> It could be a good option, yes ;-)

So I think it is a direct replacement in all occurences in these examples, though reading the rml.io specs was a little unclear about parent and children, and I feel I may have trippled up here.

Could you give this a review please as I think I have addressed your comments?

**Comment by dachafra** [comment]
If the source of the parentTriplesMap and the actual TriplesMap is the same, you do not need to put a join-condition

**Review by dachafra — APPROVED** [review]
ready to merge

**Pull request merged**